### PR TITLE
Python client

### PR DESCRIPTION
**Pull request by rabstejnek**
Create a REST client in Python.

- [x] Assessment downloads
  - [x] Create
    - [x] Literature review
    - [x] Study evaluation report
    - [x] Animal bioassay data
    - [x] Epidemiology data
    - [x] Epidemiology meta-analysis data
    - [x] In-vitro data
    - [x] Visualizations
  - [x] Purge Legacy
    - [x] Literature review
    - [x] Study evaluation report
    - [x] Animal bioassay data
    - [x] Epidemiology data
    - [x] Epidemiology meta-analysis data
    - [x] In-vitro data
    - [x] Visualizations

### CODE REVIEW

**Comment by rabstejnek** [comment]
@shapiromatron Ready for review

File: hawc/client/__init__.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            "search_type": "i",
+            "source": 2,
+            "title": title,
+            "description": description,
+            "search_string": ",".join(str(id_) for id_ in ids),
+        }
+        url = f"{self.root_url}/lit/api/search/"
+        return self.post(url, payload)
+
+    def lit_tags(self, assessment_id: int) -> pd.DataFrame:
```
Comment (shapiromatron) [inline hawc/client/__init__.py]: does this (and others below) return a dataframe, or does it return a dict, from response.json()?

File: hawc/client/__init__.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        Returns:
+            Session: a requests.Session object
+        """
+        session = Session()
+
+        url = f"{self.root_url}/user/api/token-auth/"
+        data = {"username": email, "password": password}
+        response = session.post(url, json=data)
+        token = self._handle_hawc_response(response)["token"]
+        print(response)
```
Comment (shapiromatron) [inline hawc/client/__init__.py]: remove print

File: hawc/apps/lit/urls.py [hunk 3]
```diff
@@ -50,11 +50,6 @@
         views.RefsByTagJSON.as_view(),
         name="refs_json",
     ),
-    url(
-        r"^assessment/(?P<pk>\d+)/references/download/$",
-        views.RefDownloadExcel.as_view(),
-        name="ref_download_excel",
```
Comment (shapiromatron) [inline hawc/apps/lit/urls.py]: looks like this is also in `reference_list.html` and `reference_visual.html` (ctrl-shift-f for ref_download_excel in vscode)

File: hawc/apps/lit/views.py [hunk 4]
```diff
@@ -76,69 +76,6 @@ def get_form_kwargs(self):
         return kwargs
 
 
-class RefDownloadExcel(BaseList):
```
Comment (shapiromatron) [inline hawc/apps/lit/views.py]: good riddance!

File: hawc/client/__init__.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        session.headers.update(Authorization=f"Token {token}")
+        return session
+
+    def get(self, url: str) -> Optional[Dict]:
```
Comment (shapiromatron) [inline hawc/client/__init__.py]: perhaps create a "private" method to `_format_to_df()` which takes the response we'll get from all of these and returns a data frame.

Also, these two methods are private, aka they shouldn't be exposed to our public api, so prefix them with a single underscore, eg., `_get()`, `_post()`, this is a pretty standard python convention for "private" stuff

File: hawc/apps/lit/api.py [hunk 6]
```diff
@@ -55,6 +56,27 @@ def reference_tags(self, request, pk):
...
+        methods=("get", "post"), url_path="references-download", renderer_classes=PandasRenderers
+    )
+    def references_download(self, request, pk):
+        """
+        Get all references in an assessment.
+        """
+
+        self.set_legacy_attr(pk)
+
+        self.tags = models.ReferenceFilterTag.get_all_tags(self.assessment.id, json_encode=False)
```
Comment (shapiromatron) [inline hawc/apps/lit/api.py]: we can just set this to a local variable; no need to bind to the class (change `self.tags =` to `tags = `)

File: hawc/apps/invitro/api.py [hunk 7]
```diff
@@ -1,6 +1,42 @@
...
+            return self.model.objects.published(self.assessment)
+        return self.model.objects.get_qs(self.assessment)
+
+    @decorators.detail_route(
+        methods=("get",), url_path="full-export", renderer_classes=PandasRenderers
+    )
+    def full_export(self, request, pk):
+        self.set_legacy_attr(pk)
+        self.object_list = self.get_queryset()
+        export_format = request.GET.get("output", "excel")
```
Comment (shapiromatron) [inline hawc/apps/invitro/api.py]: i don't think we need this; our renderer handles the export_format

File: hawc/client/__init__.py (hunk 1 above)
Comment (rabstejnek) [reply hawc/client/__init__.py]: All functions now return a dataframe.

File: hawc/client/__init__.py (hunk 2 above)
Comment (rabstejnek) [reply hawc/client/__init__.py]: Done

File: hawc/apps/lit/urls.py (hunk 3 above)
Comment (rabstejnek) [reply hawc/apps/lit/urls.py]: Updated reference_list.html and reference_visual.html to use new endpoint.

File: hawc/client/__init__.py (hunk 5 above)
Comment (rabstejnek) [reply hawc/client/__init__.py]: get and post methods were made private. I managed to use the dataframe constructor directly on the json, so it did not warrant a separate helper method.

File: hawc/apps/lit/api.py (hunk 6 above)
Comment (rabstejnek) [reply hawc/apps/lit/api.py]: Done

File: hawc/apps/invitro/api.py (hunk 7 above)
Comment (rabstejnek) [reply hawc/apps/invitro/api.py]: Understood, I just hardwired 'excel' into the export_format parameter.

**Comment by rabstejnek** [comment]
@shapiromatron Requested changes made and submitted.

File: hawc/apps/lit/api.py [hunk 8]
```diff
@@ -55,6 +56,27 @@ def reference_tags(self, request, pk):
         df = models.ReferenceTags.objects.as_dataframe(instance.id)
         return Response(df)
 
+    @decorators.detail_route(
+        methods=("get", "post"), url_path="references-download", renderer_classes=PandasRenderers
```
Comment (shapiromatron) [inline hawc/apps/lit/api.py]: this should just be get, no need to have post as well

Comment (shapiromatron) [reply hawc/apps/lit/api.py]: @rabstejnek after making this change, LGTM! Please make this change and then squash+merge. Congrats on your first PR 🎊

Comment (rabstejnek) [reply hawc/apps/lit/api.py]: Made the changes, will squash+merge now!

**Pull request merged**